### ISSUE TITLE
Volumes not mounted if there is a virtual switch present on the NAS

### ISSUE THREAD

**Issue by pkerwien** [qnap-dev/QNAP-CSI-PlugIn#8]
Hi,

When installing applications on the NAS like Virtualization Station, Ubuntu Linux Station, a virtual switch is created on the NAS. After that iSCSI volumes can no longer be mounted by the QNAP-CSI-PlugIn version 1.2.0. A LUN is created on the NAS, but it remains unmapped and the following error message can be found in the trident-controller:

```
time="2024-05-05T11:13:47Z" level=debug msg="Result of pool match for storage class." attributesMatch=true logLayer=core match=true pool=2 poolsMatch=true requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI storageClass= workflow="volume=create"
time="2024-05-05T11:13:48Z" level=warning msg="failed to create volume in pool" Error="rpc error: code = Unknown desc = Iscsi CREATE Failed : all adapters are unknown; please check if the adapters name are valid" Name=trident-pvc-0dab333c-0279-4415-b974-25517c2b01fe Pool=1 Size=5 logLayer=core requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI workflow="volume=create"
time="2024-05-05T11:13:50Z" level=warning msg="failed to create volume in pool" Error="rpc error: code = Unknown desc = Iscsi CREATE Failed : all adapters are unknown; please check if the adapters name are valid" Name=trident-pvc-0dab333c-0279-4415-b974-25517c2b01fe Pool=2 Size=5 logLayer=core requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI workflow="volume=create"
time="2024-05-05T11:13:50Z" level=warning msg="All pools are not available in Backend" Backend=TVS-h874 Vol Name=trident-pvc-0dab333c-0279-4415-b974-25517c2b01fe logLayer=core requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI workflow="volume=create"
time="2024-05-05T11:13:50Z" level=warning msg="Failed to create the volume on this pool." backend=TVS-h874 backendUUID=8d88257f-6532-490b-a317-499f7cfee891 error="failed to create volume on each pool; err : rpc error: code = Unknown desc = Iscsi CREATE Failed : all adapters are unknown; please check if the adapters name are valid" logLayer=core pool=TVS-h874-Non-Tiering requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI volume=pvc-0dab333c-0279-4415-b974-25517c2b01fe workflow="volume=create"
time="2024-05-05T11:13:50Z" level=debug msg="Volume event." eventType=Normal logLayer=csi_frontend message="encountered error(s) in creating the volume: [Failed to create volume pvc-0dab333c-0279-4415-b974-25517c2b01fe on storage pool TVS-h874-Non-Tiering from backend TVS-h874: failed to create volume on each pool; err : rpc error: code = Unknown desc = Iscsi CREATE Failed : all adapters are unknown; please check if the adapters name are valid]" name=pvc-0dab333c-0279-4415-b974-25517c2b01fe reason=ProvisioningFailed requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI workflow="volume=create"
time="2024-05-05T11:13:50Z" level=debug msg="Found cached PVC by UID." logLayer=csi_frontend name=webdata namespace=nginx requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI uid=0dab333c-0279-4415-b974-25517c2b01fe workflow="volume=create"
time="2024-05-05T11:13:50Z" level=debug msg="<<<< CreateVolume" Method=CreateVolume Type=CSI_Controller logLayer=csi_frontend name=pvc-0dab333c-0279-4415-b974-25517c2b01fe requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI workflow="volume=create"
time="2024-05-05T11:13:50Z" level=error msg="GRPC error: rpc error: code = Unknown desc = encountered error(s) in creating the volume: [Failed to create volume pvc-0dab333c-0279-4415-b974-25517c2b01fe on storage pool TVS-h874-Non-Tiering from backend TVS-h874: failed to create volume on each pool; err : rpc error: code = Unknown desc = Iscsi CREATE Failed : all adapters are unknown; please check if the adapters name are valid]" logLayer=csi_frontend requestID=9f61682a-5715-4470-83c6-491d561f3e5a requestSource=CSI

```

I have tested with the following firmwares, all resulting in the same result:

- QTS 5.0.1.2514
- QTS 5.1.5.2679
- QTS 5.1.6.2722
- QTS 5.2.0.2737 Beta

If I uninstall the applications that require a virtual switch and remove the virtual switch, everything starts to work again.

My backend config looks like this:

```
apiVersion: trident.qnap.io/v1
kind: TridentBackendConfig
metadata:
  name: backend-config
  namespace: trident
spec:
  version: 1
  storageDriverName: qnap-iscsi
  backendName: TVS-h874
  networkInterfaces: []
  credentials:
    name: backend-secret
  debugTraceFlags:
    method: false
  storage:
    - labels:
        storage: TVS-h874
        serviceLevel: Any
    - labels:
        performance: basic
      features:
        tiering: Disable
      serviceLevel: Non-Tiering
```

**Comment by davidcheng0716** [comment]
Hi, Thank you for your comment,

Can you help me to set the adapter name in the backendconfig, update the backend and try it one more time ? 

![Screenshot from 2024-05-06 10-19-58](https://github.com/qnap-dev/QNAP-CSI-PlugIn/assets/6870488/f97f4ecf-9708-4565-8d21-0592e834b756)

you can find the adapter information in Control Panel -> Network & Virtual Switch -> Interfaces (choose the ip you set in backend)

In above case, 
networkInterfaces: ["Adapter 1"] (ignore the words (1GbE))

**Comment by pkerwien** [comment]
@davidcheng0716 

Hi, if I set my adapter name (in my case Adapter 1) in the backend config, it comes a little bit further. Then the LUN gets mapped to a target, but its never get mounted into the pod. I can now see the following logs in the trident-node-linux pod.

My new backend config:

```
apiVersion: trident.qnap.io/v1
kind: TridentBackendConfig
metadata:
  name: backend-config
  namespace: trident
spec:
  version: 1
  storageDriverName: qnap-iscsi
  backendName: TVS-h874
  networkInterfaces: ["Adapter 1"]
  credentials:
    name: backend-secret
  debugTraceFlags:
    method: false
  storage:
    - labels:
        storage: TVS-h874
        serviceLevel: Any
    - labels:
        performance: basic
      features:
        tiering: Disable
      serviceLevel: Non-Tiering
```

Logs from the trident-node-linux pod:

```
time="2024-05-06T04:42:36Z" level=debug msg="REST API call received." Duration="18.53µs" Method=GET RequestURL=/liveness Route=LivenessProbe logLayer=rest_frontend requestID=e311aeb2-9b51-4c08-a265-95be8d31f301 requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:36Z" level=debug msg="REST API call complete." Duration="187.597µs" Method=GET RequestURL=/liveness Route=LivenessProbe StatusCode=200 logLayer=rest_frontend requestID=e311aeb2-9b51-4c08-a265-95be8d31f301 requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< osutils.execCommandWithTimeoutAndInput." command=iscsiadm error="exit status 11" logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< osutils.execCommandWithTimeout." logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=error msg="Failed to discover targets" error="exit status 11" logLayer=csi_frontend output="iscsiadm: cannot make connection to 192.168.0.50: Connection refused\niscsiadm: cannot make connection to 192.168.0.50: Connection refused\niscsiadm: cannot make connection to 192.168.0.50: Connection refused\niscsiadm: cannot make connection to 192.168.0.50: Connection refused\niscsiadm: cannot make connection to 192.168.0.50: Connection refused\niscsiadm: cannot make connection to 192.168.0.50: Connection refused\niscsiadm: connection login retries (reopen_max) 5 exceeded\niscsiadm: Could not perform SendTargets discovery: iSCSI PDU timed out\n" portal="192.168.0.50:3260" requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< iscsi.ensureIscsiTarget" logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=error msg="unable to ensure iSCSI target exists: failed to discover targets: exit status 11" err="failed to discover targets: exit status 11" iface=default logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI targetIqn=iscsi-trident-pvc-151448df-c174-4453-925a-fe94ccaf19a7 tp="192.168.0.50:3260" workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< iscsi.EnsureISCSISessions" logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< iscsi.AttachISCSIVolume" logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< iscsi.AttachISCSIVolumeRetry" logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="<<<< NodeStageVolume" Method=NodeStageVolume Type=CSI_Node logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI workflow="node_server=stage"
time="2024-05-06T04:42:39Z" level=debug msg="Released shared lock (NodeStageVolume-pvc-151448df-c174-4453-925a-fe94ccaf19a7)." lock=csi_node_server logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI
time="2024-05-06T04:42:39Z" level=error msg="GRPC error: rpc error: code = Internal desc = failed to stage volume: iSCSI login failed" logLayer=csi_frontend requestID=211a3e6d-4a91-4ca6-af9e-c1c3c134b537 requestSource=CSI
time="2024-05-06T04:42:46Z" level=debug msg="REST API call received." Duration="25.951µs" Method=GET RequestURL=/readiness Route=ReadinessProbe logLayer=rest_frontend requestID=0c8a4ebe-c805-4ab5-8ff4-f7aea6bc884a requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:46Z" level=debug msg="REST API call complete." Duration="216.869µs" Method=GET RequestURL=/readiness Route=ReadinessProbe StatusCode=200 logLayer=rest_frontend requestID=0c8a4ebe-c805-4ab5-8ff4-f7aea6bc884a requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:46Z" level=debug msg="REST API call received." Duration="11.5µs" Method=GET RequestURL=/liveness Route=LivenessProbe logLayer=rest_frontend requestID=bc582ec8-90e8-4411-8af5-e8909797f022 requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:46Z" level=debug msg="REST API call complete." Duration="127.625µs" Method=GET RequestURL=/liveness Route=LivenessProbe StatusCode=200 logLayer=rest_frontend requestID=bc582ec8-90e8-4411-8af5-e8909797f022 requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:56Z" level=debug msg="REST API call received." Duration="15.76µs" Method=GET RequestURL=/readiness Route=ReadinessProbe logLayer=rest_frontend requestID=21f5a0bf-942f-4479-b00f-41db8ab3be5f requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:56Z" level=debug msg="REST API call complete." Duration="155.076µs" Method=GET RequestURL=/readiness Route=ReadinessProbe StatusCode=200 logLayer=rest_frontend requestID=21f5a0bf-942f-4479-b00f-41db8ab3be5f requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:56Z" level=debug msg="REST API call received." Duration="9.261µs" Method=GET RequestURL=/liveness Route=LivenessProbe logLayer=rest_frontend requestID=aefb966b-0cf9-4011-a5a6-ecc42ccace50 requestSource=REST workflow="trident_rest=logger"
time="2024-05-06T04:42:56Z" level=debug msg="REST API call complete." Duration="121.356µs" Method=GET RequestURL=/liveness Route=LivenessProbe StatusCode=200 logLayer=rest_frontend requestID=aefb966b-0cf9-4011-a5a6-ecc42ccace50 requestSource=REST workflow="trident_rest=logger"
```

**Comment by LeonaChen2727** [comment]
We have released v1.2.1, which fixes this issue.
Please refer to release note: https://github.com/qnap-dev/QNAP-CSI-PlugIn/releases/tag/v1.2.1

**Comment by pkerwien** [comment]
Thanks! Unfortunately I cannot test it right now, since I have now taken my NAS into "production" with QuTS hero. So I need to wait for that support before I can start testing things again.
